lib/tests/streamlit/elements/map_test.py

```diff
@@ -96,6 +96,51 @@ def test_main_kwargs(self):
         # Also test that the radius property is set up correctly.
         self.assertEqual(c.get("layers")[0].get("radiusMinPixels"), 3)
 
+    @parameterized.expand(
+        [
+            ("string_index", ["a", "b", "c"]),
+            ("indexed_from_1", [1, 2, 3]),
+        ]
+    )
+    def test_alternative_dataframe_index(self, _, index):
+        """Test that the map method does not error with non-standard dataframe indexes"""
+        df = pd.DataFrame(
+            {
+                "lat": [38.8762997, 38.8742997, 38.9025842],
+                "lon": [-77.0037, -77.0057, -77.0556545],
+                "color": [[255, 0, 0, 128], [0, 255, 0, 128], [0, 0, 255, 128]],
+                "size": [100, 50, 30],
+            },
+            index=index,
+        )
+
+        st.map(df, size="size", color="color")
+        c = json.loads(self.get_delta_from_queue().new_element.deck_gl_json_chart.json)
+
+        self.assertEqual(c.get("layers")[0].get("getFillColor"), "@@=color")
+        self.assertEqual(c.get("layers")[0].get("getRadius"), "@@=size")
+
+    def test_named_dataframe_index(self):
```

Comment:
Oh, I initially thought it was about the index name... but it's actually about the index value 😅 so your test above is the relevant one. But let's keep this one in as well.